src/functions/getIvaDefects.ts

```diff
@@ -23,14 +19,19 @@ export const getIvaDefects: Handler = async (
     return addHttpHeaders(defectErrors);
   }
 
-  const vehicleType = event?.queryStringParameters?.vehicleType as VehicleType;
-  const euVehicleCategory = event?.queryStringParameters
+  const euVehicleCategoryQuery = event?.queryStringParameters
     ?.euVehicleCategory as EUVehicleCategory;
-  const inspectionType = event?.queryStringParameters
-    ?.inspectionType as InspectionType;
+
+  if (!euVehicleCategoryQuery) {
+    return new HTTPResponse(400, "euVehicleCategory required");
+  } else if (
+    !Object.values(EUVehicleCategory).includes(euVehicleCategoryQuery)
+  ) {
+    return new HTTPResponse(204, "[]");
```

Comment:
Looking closer at this, and in light of discussions with @cb-cs... 204 is definitely not the right response for this.
we're returning a 204 to suggest we have successfully searched the database and found no results - but we are returning it in response to an incorrect query parameter - before ever attempting a search on the data